packages/react/src/chakra-composable-provider.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react"
+import {
+  ChakraProvider as BaseChakraProvider,
+  ChakraProviderProps as BaseChakraProviderProps,
+} from "@chakra-ui/provider"
+import { theme } from "@chakra-ui/theme"
+import { ToastProvider, ToastProviderProps } from "@chakra-ui/toast"
+import { FramerMotionProvider } from "./framer-motion-provider"
+import { AnimatePresence, motion } from "framer-motion"
+
+type ChakraFeature = {
+  id: string
+  Provider: React.ComponentType<React.PropsWithChildren<{}>>
+}
+
+export function createToastFeature(
+  toastOptions?: ToastProviderProps,
+): ChakraFeature {
+  return {
+    id: "toast",
+    Provider(props) {
+      return <ToastProvider {...props} {...toastOptions} />
+    },
+  }
+}
```

Comment:
This is an independent function so that the reference to `ToastProvider` can be dropped by tree-shaking.